sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -3740,3 +3743,147 @@ def hpat_pandas_series_pct_change_impl(self, periods=1, fill_method='pad', limit
 
     return hpat_pandas_series_pct_change_impl
+
+
+@overload_method(SeriesType, 'describe')
+def hpat_pandas_series_describe(self, percentiles=None, include=None, exclude=None):
+    """
+    Pandas Series method :meth:`pandas.Series.describe` implementation.
+
+    Note: Differs from Pandas in returning statistics as Series of strings when applied to
+        Series of strings or date-time values
+
+    .. only:: developer
+
+       Tests: python -m sdc.runtests -k sdc.tests.test_series.TestSeries.test_series_describe*
+
+    Parameters
+    ----------
+    self: :obj:`pandas.Series`
+        Input series
+    percentiles: :obj:`list-like`
+        The percentiles to include in the output. The default is [.25, .5, .75]
+        All should fall between 0 and 1 and no duplicates are allowed.
+    include: 'all', :obj:`list-like` of dtypes or None, default None
+        A white list of data types to include in the result. Ignored for Series.
+    exclude: :obj:`list-like` of dtypes or None, default None
+        A black list of data types to omit from the result. Ignored for Series.
+
+    Returns
+    -------
+    :obj:`pandas.Series`
+        returns :obj:`pandas.Series` object containing summary statistics of the Series
+    """
+
+    ty_checker = TypeChecker('Method describe().')
+    ty_checker.check(self, SeriesType)
+
+    if not (isinstance(percentiles, (types.List, types.Array, types.UniTuple))
+            and isinstance(percentiles.dtype, types.Number)
+            or isinstance(percentiles, (types.Omitted, types.NoneType))
+            or percentiles is None):
+        ty_checker.raise_exc(percentiles, 'list-like', 'percentiles')
+
+    is_percentiles_none = percentiles is None or isinstance(percentiles, (types.Omitted, types.NoneType))
+
+    if isinstance(self.dtype, types.Number):
+        def hpat_pandas_series_describe_numeric_impl(self, percentiles=None, include=None, exclude=None):
+
+            if is_percentiles_none == False:  # noqa
+                percentiles_list = list(percentiles)
+                median_in_percentiles = 0.5 in percentiles_list
+                if not median_in_percentiles:
+                    percentiles_list.append(0.5)
+                sorted_percentiles = sorted(percentiles_list)
+
+                # check percentiles have correct values:
+                arr = numpy.asarray(sorted_percentiles)
+                if len(numpy.unique(arr)) != len(arr):
+                    raise ValueError("percentiles cannot contain duplicates")
+                if numpy.any(arr[(arr < 0) * (arr > 1)]):
+                    raise ValueError("percentiles should all be in the interval [0, 1].")
+
+                # TODO: support proper rounding of percentiles like in pandas.io.formats.format.format_percentiles
+                # requires numpy.round(precision), numpy.isclose to be supported by Numba
+                percentiles_indexes = []
+                for percentile in arr:
```

Comment:
I'd move this functionality into separate function

Reply:
Moved into common_functions._sdc_pandas_format_percentiles